lib/vagrant-service-manager/action/restart_service.rb

```diff
@@ -0,0 +1,34 @@
+module Vagrant
+  module ServiceManager
+    module Action
+      class RestartService
+        def initialize(app, env)
+          @app = app
+          @machine = env[:machine]
+          @ui = env[:ui]
+          @providers = @machine.config.servicemanager.providers.split(',').map(&:chomp)
+        end
+
+        def execute_command(command, provider)
+           @machine.communicate.execute(command) do |type, data|
+            if type == :stderr
+              @ui.error(data)
+            else
+              @ui.info("# Restarted #{provider} service.")
+            end
+          end
+        end
+
+        def call(env)
+          @providers.each do |provider|
+            command = "sudo systemctl restart #{provider}"
```

Comment:
I feel like each provider should have it's own module that defines restart for it.  This way we don't have to worry about someone passing in unexpected values.  i.e. vagrant service-manager restart httpd - if we dont' have apache defined.

This also may make it cleaner to add new providers down the road.  

for POC this is OK, but not for merge.